service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ImmunizationObservationStatementMapper.java

```diff
@@ -0,0 +1,265 @@
+        }
+        throw new EhrMapperException("Could not map recorded date");
+    }
+
+    private String buildEffectiveTime(Immunization immunization) {
+        immunization.getDateElement();
+        immunization.getDate();
+        Optional<String> effectiveTime = Optional.empty();
+        if (immunization.getDate() != null) {
+            effectiveTime = Optional.of(DateFormatUtil.formatDate(immunization.getDate()));
+        }
+        return effectiveTime.orElse(StringUtils.EMPTY);
+    }
+
+    private String buildPertinentInformation(Immunization immunization, Bundle bundle) {
+        List<String> pertinentInformationList = retrievePertinentInformation(immunization, bundle);
+        return pertinentInformationList.stream()
+            .filter(StringUtils::isNotEmpty)
+            .collect(Collectors.joining("\n"));
+    }
+
+    private List<String> retrievePertinentInformation(Immunization immunization, Bundle bundle) {
+        return List.of(
+            buildParentPresentPertinentInformation(immunization),
+            buildLocationPertinentInformation(immunization, bundle),
+            buildManufacturerPertinentInformation(immunization, bundle),
+            buildLotNumberPertinentInformation(immunization),
+            buildExpirationDatePertinentInformation(immunization),
+            buildSitePertinentInformation(immunization),
+            buildRoutePertinentInformation(immunization),
+            buildDoseQuantityPertinentInformation(immunization),
+            buildNotePertinentInformation(immunization),
+            buildExplanationPertinentInformation(immunization),
+            buildVaccinationProtocolPertinentInformation(immunization)
+        );
+    }
+
+    private String buildParentPresentPertinentInformation(Immunization immunization) {
+        var parentPresentOptional = filterExtensionByUrl(immunization, PARENT_PRESENT_URL);
+        if (parentPresentOptional.isPresent()) {
+            BooleanType isParentPresent = (BooleanType) parentPresentOptional.get().getValue();
+            return PARENT_PRESENT + isParentPresent.getValue();
+        }
+        return StringUtils.EMPTY;
+    }
+
+    private String buildLocationPertinentInformation(Immunization immunization, Bundle bundle) {
+        Optional<Location> location = Optional.empty();
+        if (immunization.getLocation() != null) {
+            String locationId = immunization.getLocation().getReferenceElement().getIdPart();
+            location = bundle.getEntry().stream()
+                .filter(bundleEntryComponent -> bundleEntryComponent.getResource().getResourceType().equals(ResourceType.Location))
+                .map(bundleEntryComponent -> (Location) bundleEntryComponent.getResource())
+                .filter(bundleLocation -> bundleLocation.getIdElement().getIdPart().equals(locationId))
+                .findFirst();
+        }
+
+        return location.map(value -> LOCATION + value.getName()).orElse(StringUtils.EMPTY);
```

Comment:
You can use similar construction up. I would recommend putting `orElse` in a new line.